### PR TITLE
DKC3 v1.1.0

### PR DESCRIPTION
**Pull request by PoryGone**
Changelist:

Features:
- KONGsanity option (Collect all KONG letters in each level for a check)
- Autosave option
- Difficulty option
- MERRY option
- Handle collected/co-op locations

Bugfixes:
- Fixed Mekanos softlock
- Prevent Brothers Bear giving extra Banana Birds
- Fixed Banana Bird Mother check sending prematurely
- Fix Logic bug with Krematoa level costs

Please Squash

### CODE REVIEW

File: worlds/dkc3/__init__.py [hunk 1]
```diff
@@ -118,7 +121,13 @@ def generate_basic(self):
 
         total_junk_count = total_required_locations - len(itempool)
 
-        itempool += [self.create_item(ItemName.bear_coin)] * total_junk_count
+        junk_pool = []
+        junk_keys = list(junk_table.keys())
```
Comment (Berserker66) [inline worlds/dkc3/__init__.py:125]: https://docs.python.org/3/library/random.html#random.choices (random.choices) should allow you to convert this loop into a single call, example exists in Subnautica.

File: worlds/dkc3/Regions.py [hunk 2]
```diff
@@ -44,6 +44,10 @@ def create_regions(world, player: int, active_locations):
         LocationName.lakeside_limbo_bonus_2 : [0x657, 3],
         LocationName.lakeside_limbo_dk      : [0x657, 5],
     }
+    if world.kongsanity[player]:
+        lakeside_limbo_region_locations.update({
```
Comment (Berserker66) [inline worlds/dkc3/Regions.py:48]: isn't a dealbraker, but more clean would be lakeside_limbo_region_locations[LocationName.lakeside_limbo_kong] = [] for these

File: worlds/dkc3/Options.py [hunk 3]
```diff
@@ -75,13 +75,55 @@ class PercentageOfBananaBirds(Range):
...
+
+    @classmethod
+    def get_option_name(cls, value) -> str:
+        if cls.auto_display_name:
+            return cls.name_lookup[value].upper()
+        else:
+            return cls.name_lookup[value]
+
+
+class Autosave(DefaultOnToggle):
```
Comment (Berserker66) [inline worlds/dkc3/Options.py:113]: Is there ever a reason to not have autosave? as in, why is this an option?

Comment (PoryGone) [reply worlds/dkc3/Options.py:113]: There has been a single case of a softlock with this randomizer, which would have been permanent if autosave had been on. That is fixed and there hopefully isn't another, but that is one concern.
Also, the game calls you a "Cheatin' Chump" at the end if you use the autosave, which some players may not want.

File: worlds/dkc3/Regions.py (hunk 2 above)
Comment (PoryGone) [reply worlds/dkc3/Regions.py:48]: Unless this is a performance hit doing it this way, I think for file consistency I'd rather leave it this way for now. I may go through this file for cleanup in future, as it is doing a lot of stuff it doesn't need to anymore, but I don't want to break everything right this moment.

Comment (Berserker66) [reply worlds/dkc3/Regions.py:48]: There likely is a performance hit, as you're creating a dict and then merging two dicts; as opposed to just adding an entry to an existing dict. So there's way more memory shuffling going around, at least one extra container and such.

Comment (PoryGone) [reply worlds/dkc3/Regions.py]: OK, I have resolved this.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**